### ISSUE TITLE
nnUNet_convert_decathlon_task error

### ISSUE THREAD

**Issue by sungchu** [MIC-DKFZ/nnUNet#510]
When I execute the following command:
nnUNet_convert_decathlon_task -i /root/user/home/qiao/nnUNetFrame/DATASET/nnUNet_raw/nnUNet_raw_data/Task58_Brain/
I didn't find decathlon folder under nnUNet_raw_data folder. But I also didn't get any error.

When I execute the following command:
`ls /root/user/home/qiao/nnUNetFrame/DATASET/nnUNet_raw/nnUNet_raw_data/Task58_Brain`
The output is:
dataset.json  imagesTr  imagesTs  labelsTr  labelsTs

Can anyone tell me what am I doing wrong?

**Comment by FabianIsensee** [comment]
This does not look like one of the 10 decathlon tasks so there is no need to run this command

**Comment by squaresoft2015** [comment]
Same question here...I did not find any directory like $nnUNet_raw_data_base/nnUNet_raw_data/Task005_Prostate/imagesTs, but only $nnUNet_raw_data_base/nnUNet_raw_data/Task05_Prostate/imagesTs. If the conversion command is not necessary, how can I create Task005_Prostate folder? Just simply rename the folder name as it ? It still makes an error during prediction.

**Comment by sungchu** [comment]
@squaresoft2015 I just simply rename the folder. Because you already get the correct file format, there is no need to run nnUNet_convert_decathlon_task.

Thanks @FabianIsensee for help

**Comment by squaresoft2015** [comment]
When I download the pretrained model, it prompts:

Download finished. Extracting...
Archive:  /data/home/amaxcuda9/.nnunetdownload_16135322765340044
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /data/home/amaxcuda9/.nnunetdownload_16135322765340044 or
        /data/home/amaxcuda9/.nnunetdownload_16135322765340044.zip, and cannot find /data/home/amaxcuda9/.nnunetdownload_16135322765340044.ZIP, period.
Done

Is it the reason of network ? The download file is not complete? I tried several times and the error remains the same.

**Comment by FabianIsensee** [comment]
Do not just rename the folder. If you download a decathlon task you need to convert it. @sungchu did not need to convert because he was using a custom dataset.

If you have problems downloading a pretrained model, please download the zip file from zenodo directly: https://zenodo.org/record/4485926

Then use nnUNet_install_pretrained_model_from_zip to install the model

**Comment by FabianIsensee** [comment]
Please follow this tutorial: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/inference_example_Prostate.md

**Comment by squaresoft2015** [comment]
> Please follow this tutorial: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/inference_example_Prostate.md

I have run nnUNet_convert_decathlon_task -i /xxx/Task05_Prostate, there is no error message and I have not seen any generated Task005_Prostate folder in the corresponding directory. Is there anything wrong?

**Comment by FabianIsensee** [comment]
the converted task will be in the nnUNet_raw_data_base/nnUNet_raw_data folder. nnUNet_raw_data_base is the environment variable you needed to set when  installing nnunet

**Comment by squaresoft2015** [comment]
Thank you for your kind and timely reply! I use --save_npz during prediction and would like to output the softmax probabilities for ensembling. But the output folder only contains *.nii,gz, *.pkl and *.jason. Where I can find  the probabilities?

**Comment by FabianIsensee** [comment]
If you ran nnUNet_predict with `--save_npz` then there should also be .npz files in the output directory. These are the softmax outputs.

nnUNet_ensemble does not generate ensembled probability outputs.

Best,
Fabian

**Comment by squaresoft2015** [comment]
The command I ran is : `nnUNet_predict -i nnUNet_raw_data_base/nnUNet_raw_data/Task005_Prostate/imagesTs/ -o segmentation_result/ -t 5 --save_npz -m 3d_fullres`
 and the out put is :  using model stored in  nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1
This model expects 2 input modalities for each image
Found 16 unique case ids, here are some examples: ['prostate_11' 'prostate_30' 'prostate_09' 'prostate_23' 'prostate_33'
 'prostate_12' 'prostate_33' 'prostate_27' 'prostate_12' 'prostate_03']
If they don't look right, make sure to double check your filenames. They must end with _0000.nii.gz etc
number of cases: 16
number of cases that still need to be predicted: 0
emptying cuda cache
loading parameters for folds, None
folds is None so we will automatically look for output folders (not using 'all'!)
found the following folds:  ['nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_0', 'nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_1', 'nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_2', 'nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_3', 'nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_4']
2021-02-22 20:00:33.431776: Using dummy2d data augmentation
using the following model files:  ['nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_0/model_final_checkpoint.model', 'nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_1/model_final_checkpoint.model', 'nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_2/model_final_checkpoint.model', 'nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_3/model_final_checkpoint.model', 'nnUNet_trained_models/nnUNet/3d_fullres/Task005_Prostate/nnUNetTrainerV2__nnUNetPlansv2.1/fold_4/model_final_checkpoint.model']
starting preprocessing generator
starting prediction...
inference done. Now waiting for the segmentation export to finish...
postprocessing...

I did not see the *.npz file in the results folder.

**Comment by FabianIsensee** [comment]
I did exactly what you did and the files are created properly:

![image](https://user-images.githubusercontent.com/11302164/108711299-6e496980-7515-11eb-875e-343203a84088.png)

(`nnUNet_predict -i imagesTs/ -o ~/deletene -t 5 --save_npz -m 3d_fullres`)

This was on the current master. I don't know how I can help you if I cannot reproduce it :-/

**Comment by squaresoft2015** [comment]
I deleted all the output files in the segmentation_result directory and re-ran the command, now I get the *.npz files. Thank you!

**Comment by FabianIsensee** [comment]
Ah that's because it checks whether the nii gz files are there but does not check for npz. You could also have used `--overwrite_existing`. But this is in fact an error by nnU-Net. Let me fix that
